Matek H743 build failed on master

## Bug report

**Issue details**
https://github.com/ArduPilot/ardupilot/pull/19706/commits/83cb5c96b045e9b35988b74c80971a1c82be5802 add following code 
https://github.com/ArduPilot/ardupilot/blob/d9c562cfae63281f1c14572c84a96cfc440902d0/libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py#L1102-L1105
In Matek H743, it reduce `flash_size` from 2048KB to 1792KB (`offset`), substract 256KB from 2MB flash
But later in origin code
https://github.com/ArduPilot/ardupilot/blob/d9c562cfae63281f1c14572c84a96cfc440902d0/libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py#L1128
`flash_reserve_end` is 256KB. So `flash_length` is 1408 which is incorrect. And it cause build failed
[813/816] Linking build/MatekH743/bin/arducopter
/opt/gcc-arm-none-eabi-10-2020-q4-major/bin/../lib/gcc/arm-none-eabi/10.2.1/../../../../arm-none-eabi/bin/ld:common.ld:201 cannot move location counter backwards (from 0000000008198200 to 0000000008180000)
collect2: error: ld returned 1 exit status

**Version**
master

**Platform**
All

**Hardware type**
Matek H743

--- a/libraries/AP_HAL_ChibiOS/hwdef/BeastH7/hwdef-bl.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/BeastH7/hwdef-bl.dat
@@ -15,8 +15,9 @@ FLASH_SIZE_KB 2048
 # bootloader starts at zero offset
 FLASH_RESERVE_START_KB 0
 
-# reserve space for flash storage in last 2 sectors
-FLASH_RESERVE_END_KB 256
+# use last 2 pages for flash storage
+# H743 has 16 pages of 128k each
+STORAGE_FLASH_PAGE 14
 
 # the location where the bootloader will put the firmware
 # the H743 has 128k sectors

--- a/libraries/AP_HAL_ChibiOS/hwdef/BeastH7/hwdef.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/BeastH7/hwdef.dat
@@ -110,9 +110,6 @@ define HAL_STORAGE_SIZE 16384
 # H743 has 16 pages of 128k each
 STORAGE_FLASH_PAGE 14
 
-# reserve space for flash storage in last 2 sectors
-FLASH_RESERVE_END_KB 256
-
 # enable logging to dataflash
 define HAL_LOGGING_DATAFLASH_ENABLED 1
 

--- a/libraries/AP_HAL_ChibiOS/hwdef/KakuteH7/hwdef-bl.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/KakuteH7/hwdef-bl.dat
@@ -18,8 +18,8 @@ FLASH_RESERVE_START_KB 0
 # the location where the bootloader will put the firmware
 FLASH_BOOTLOADER_LOAD_KB 128
 
-# reserve space for flash storage in last 2 sectors
-FLASH_RESERVE_END_KB 256
+STORAGE_FLASH_PAGE 14
+define HAL_STORAGE_SIZE 32768
 
 # order of UARTs (and USB)
 SERIAL_ORDER OTG1

--- a/libraries/AP_HAL_ChibiOS/hwdef/KakuteH7/hwdef.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/KakuteH7/hwdef.dat
@@ -24,9 +24,6 @@ FLASH_RESERVE_START_KB 128
 STORAGE_FLASH_PAGE 14
 define HAL_STORAGE_SIZE 32768
 
-# reserve space for flash storage in last 2 sectors
-FLASH_RESERVE_END_KB 256
-
 # one I2C bus
 I2C_ORDER I2C1
 

--- a/libraries/AP_HAL_ChibiOS/hwdef/KakuteH7Mini/hwdef-bl.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/KakuteH7Mini/hwdef-bl.dat
@@ -18,8 +18,8 @@ FLASH_RESERVE_START_KB 0
 # the location where the bootloader will put the firmware
 FLASH_BOOTLOADER_LOAD_KB 128
 
-# reserve space for flash storage in last 2 sectors
-FLASH_RESERVE_END_KB 256
+STORAGE_FLASH_PAGE 14
+define HAL_STORAGE_SIZE 32768
 
 # order of UARTs (and USB)
 SERIAL_ORDER OTG1

--- a/libraries/AP_HAL_ChibiOS/hwdef/KakuteH7Mini/hwdef.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/KakuteH7Mini/hwdef.dat
@@ -27,9 +27,6 @@ define HAL_STORAGE_SIZE 32768
 # enable logging to dataflash
 define HAL_LOGGING_DATAFLASH_ENABLED 1
 
-# reserve space for flash storage in last 2 sectors
-FLASH_RESERVE_END_KB 256
-
 # one I2C bus
 I2C_ORDER I2C1
 

--- a/libraries/AP_HAL_ChibiOS/hwdef/MatekH743/hwdef-bl.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/MatekH743/hwdef-bl.dat
@@ -15,8 +15,9 @@ FLASH_SIZE_KB 2048
 # bootloader starts at zero offset
 FLASH_RESERVE_START_KB 0
 
-# reserve space for flash storage in last 2 sectors
-FLASH_RESERVE_END_KB 256
+# use last 2 pages for flash storage
+# H743 has 16 pages of 128k each
+STORAGE_FLASH_PAGE 14
 
 # the location where the bootloader will put the firmware
 # the H743 has 128k sectors

--- a/libraries/AP_HAL_ChibiOS/hwdef/MatekH743/hwdef.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/MatekH743/hwdef.dat
@@ -180,9 +180,6 @@ define HAL_STORAGE_SIZE 16384
 # H743 has 16 pages of 128k each
 STORAGE_FLASH_PAGE 14
 
-# reserve space for flash storage in last 2 sectors
-FLASH_RESERVE_END_KB 256
-
 # spi devices
 SPIDEV mpu6000   SPI1 DEVID1 IMU1_CS      MODE3  1*MHZ  4*MHZ
 SPIDEV icm20602  SPI4 DEVID1 IMU2_CS      MODE3  1*MHZ  4*MHZ

--- a/libraries/AP_HAL_ChibiOS/hwdef/ZubaxGNSS/hwdef-bl.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/ZubaxGNSS/hwdef-bl.dat
@@ -26,9 +26,9 @@ define CH_CFG_ST_FREQUENCY 1000
 
 FLASH_SIZE_KB 256
 
-# reserve space for params
-FLASH_RESERVE_END_KB 2
-
+# store parameters in last 2 pages
+STORAGE_FLASH_PAGE 126
+define HAL_STORAGE_SIZE 800
 
 # USART3 for debug
 #STDOUT_SERIAL SD3

--- a/libraries/AP_HAL_ChibiOS/hwdef/ZubaxGNSS/hwdef.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/ZubaxGNSS/hwdef.dat
@@ -16,9 +16,6 @@ FLASH_RESERVE_START_KB 34
 STORAGE_FLASH_PAGE 126
 define HAL_STORAGE_SIZE 800
 
-# reserve space for flash storage in last 2 sectors
-FLASH_RESERVE_END_KB 2
-
 # board ID for firmware load
 APJ_BOARD_ID 1005
 
@@ -31,10 +28,6 @@ OSCILLATOR_HZ 16000000
 define CH_CFG_ST_FREQUENCY 1000
 
 FLASH_SIZE_KB 256
-
-
-# reserve space for params
-FLASH_RESERVE_END_KB 2
 
 # USART3 for debug
 STDOUT_SERIAL SD3

--- a/libraries/AP_HAL_ChibiOS/hwdef/mindpx-v2/hwdef.dat
+++ b/libraries/AP_HAL_ChibiOS/hwdef/mindpx-v2/hwdef.dat
@@ -20,9 +20,6 @@ FLASH_SIZE_KB 2048
 # space to reserve for bootloader and storage at start of flash
 FLASH_RESERVE_START_KB 16
 
-# space to reserve for storage at end of flash
-FLASH_RESERVE_END_KB 0
-
 # serial port for stdout disabled, uses USB instead
 # STDOUT_SERIAL SD7
 # STDOUT_BAUDRATE 57600

